R/pkg/R/DataFrame.R

```diff
@@ -1357,17 +1357,46 @@ setMethod("mutate",
           function(.data, ...) {
             x <- .data
             cols <- list(...)
-            stopifnot(length(cols) > 0)
+            if (length(cols) <= 0) {
+              return(x)
+            }
+
             stopifnot(class(cols[[1]]) == "Column")
+
+            # Check if there is any duplicated column name in the DataFrame
+            dfCols <- columns(x)
+            if (length(unique(dfCols)) != length(dfCols)) {
+              stop("Error: found duplicated column name in the DataFrame")
+            }
+
+            # TODO: simplify the implementation of this method after SPARK-12225 is resolved.
+
+            # The last column of the same name in the specific columns takes effect
             ns <- names(cols)
```

Comment:
no. dfCols are the column names of the dataframe, ns are the names of the columns to be added or replaced.